pkg/registry/flowcontrol/prioritylevelconfiguration/strategy.go

```diff
@@ -0,0 +1,102 @@
+var Strategy = priorityLevelConfigurationStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+
+// NamespaceScoped returns false because all PriorityClasses are global.
+func (priorityLevelConfigurationStrategy) NamespaceScoped() bool {
+	return false
+}
+
+// PrepareForCreate clears the status of a priority-level-configuration before creation.
+func (priorityLevelConfigurationStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+	pl := obj.(*flowcontrol.PriorityLevelConfiguration)
+	pl.Status = flowcontrol.PriorityLevelConfigurationStatus{}
+	pl.Generation = 1
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (priorityLevelConfigurationStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newPriorityLevelConfiguration := obj.(*flowcontrol.PriorityLevelConfiguration)
+	oldPriorityLevelConfiguration := old.(*flowcontrol.PriorityLevelConfiguration)
+
+	// Spec updates bump the generation so that we can distinguish between status updates.
+	if !apiequality.Semantic.DeepEqual(newPriorityLevelConfiguration.Spec, oldPriorityLevelConfiguration.Spec) {
+		newPriorityLevelConfiguration.Generation = oldPriorityLevelConfiguration.Generation + 1
+	}
+	newPriorityLevelConfiguration.Status = oldPriorityLevelConfiguration.Status
+}
+
+// Validate validates a new priority-level.
+func (priorityLevelConfigurationStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
+	return validation.ValidatePriorityLevelConfiguration(obj.(*flowcontrol.PriorityLevelConfiguration))
+}
+
+// Canonicalize normalizes the object after validation.
+func (priorityLevelConfigurationStrategy) Canonicalize(obj runtime.Object) {
+}
+
+func (priorityLevelConfigurationStrategy) AllowUnconditionalUpdate() bool {
+	return true
+}
+
+// AllowCreateOnUpdate is false for priority-level-configurations; this means a POST is needed to create one.
+func (priorityLevelConfigurationStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+// ValidateUpdate is the default update validation for an end user.
+func (priorityLevelConfigurationStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return validation.ValidatePriorityLevelConfiguration(obj.(*flowcontrol.PriorityLevelConfiguration))
+}
+
+type priorityLevelConfigurationStatusStrategy struct {
+	priorityLevelConfigurationStrategy
+}
+
+// StatusStrategy is the default logic that applies when updating priority level configuration objects' status.
+var StatusStrategy = priorityLevelConfigurationStatusStrategy{Strategy}
+
+func (priorityLevelConfigurationStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newPriorityLevelConfiguration := obj.(*flowcontrol.PriorityLevelConfiguration)
+	oldPriorityLevelConfiguration := old.(*flowcontrol.PriorityLevelConfiguration)
+	newPriorityLevelConfiguration.Spec = oldPriorityLevelConfiguration.Spec
```

Comment:
also clear metadata.  We only want to set status.

Reply:
good catch!